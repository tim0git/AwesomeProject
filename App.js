import 'react-native-gesture-handler';
import React from 'react';
import {NavigationContainer} from '@react-navigation/native';
import {createBottomTabNavigator} from '@react-navigation/bottom-tabs';
import Hello from './components/Hello';
import Details from './components/Details';
import GalleryNav from './components/GalleryNav';

const Tab = createBottomTabNavigator();

const App = () => (
  <NavigationContainer>
    <Tab.Navigator>
      <Tab.Screen name="Home" component={Hello} />
      <Tab.Screen name="Image Gallery" component={GalleryNav} />
      <Tab.Screen name="Details" component={Details} />
    </Tab.Navigator>
  </NavigationContainer>
);

export default App;
