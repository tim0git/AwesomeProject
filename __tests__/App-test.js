import 'react-native';
import React from 'react';
import 'jest-enzyme';
import 'react-native-mock-render/mock';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import App from '../App';

// import Hello from '../components/Hello';
// import Details from '../components/Details';
// import GalleryNav from '../components/GalleryNav';

describe('<App >', () => {
  const app = <App />;

  it('Component Renders', () => {
    const component = shallow(app);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('has a bottom nav bar', () => {
    const wrapper = shallow(app);
    expect(wrapper.find('BottomTabNavigator').at(0)).toExist();
  });

  it('bottom nav bar has 3 navigation screens', () => {
    const count = [0, 1, 2];
    const wrapper = shallow(app);
    count.forEach((number) => {
      expect(wrapper.find('Screen').at(number)).toExist();
    });
  });

  it('bottom nav bar has only 3 navigation screens', () => {
    const wrapper = shallow(app);
    expect(wrapper.find('Screen').at(3).exists()).toBe(false);
  });

  it('first screen has text Home', () => {
    const wrapper = shallow(app);
    expect(wrapper.find('Screen').first().props().name).toBe('Home');
  });

  it('index 1 (middle) screen has text Image Gallery', () => {
    const wrapper = shallow(app);
    expect(wrapper.find('Screen').at(1).props().name).toBe('Image Gallery');
  });

  it('last screen has text Details', () => {
    const wrapper = shallow(app);
    expect(wrapper.find('Screen').last().props().name).toBe('Details');
  });

  test.todo(
    'should render a <Hello > component when you press Hello on the nav bar',
  );
  test.todo(
    'should render a <ImageGallery > component when you press Image Gallery on the nav bar',
  );
  test.todo(
    'should render a <Details > component when you press Details on the nav bar',
  );
});
