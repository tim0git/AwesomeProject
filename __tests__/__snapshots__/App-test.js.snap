// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App > Component Renders 1`] = `
<ForwardRef(NavigationContainer)>
  <BottomTabNavigator>
    <Screen
      component={[Function]}
      name="Home"
    />
    <Screen
      component={[Function]}
      name="Image Gallery"
    />
    <Screen
      component={[Function]}
      name="Details"
    />
  </BottomTabNavigator>
</ForwardRef(NavigationContainer)>
`;
